### DISCUSSION TITLE
what is a spike?

### DISCUSSION THREAD

**Discussion by oldwindmill** [amberelectric/public-api#49] [Q&A] [answered]
two intervals from todays prices. both have the same range for prices, but one is a spike, the other not. the first predicts 1735 (which is outside the range, btw)  , the second 1.35. is a spike defined as a forecast over a certain value? (ah, remembering, over $3?) 
{
    "type": "ForecastInterval",
    "date": "2021-07-05",
    "duration": 30,
    "startTime": "2021-07-05T09:00:01Z",
    "endTime": "2021-07-05T09:30:00Z",
    "nemTime": "2021-07-05T19:30:00+10:00",
    "perKwh": 1735.58433,
    "renewables": 2.867389065839971,
    "spotPerKwh": 1589.5044000000003,
    "channelType": "general",
    "spikeStatus": "spike",
    "tariffInformation": {
      "period": "peak"
    },
    "range": {
      "min": 8.49216,
      "max": 1499.998
    }
  },
  {
    "type": "ForecastInterval",
    "date": "2021-07-05",
    "duration": 30,
    "startTime": "2021-07-05T09:30:01Z",
    "endTime": "2021-07-05T10:00:00Z",
    "nemTime": "2021-07-05T20:00:00+10:00",
    "perKwh": 135.20267,
    "renewables": 2.9280016836788385,
    "spotPerKwh": 103.3807841,
    "channelType": "general",
    "spikeStatus": "none",
    "tariffInformation": {
      "period": "peak"
    },
    "range": {
      "min": 8.11664,
      "max": 1499.995
    }

**Comment by madpilot** [comment, marked as answer]
Dammit. I think the range might be ex-GST - I will fix that... (Also the range is relative to the `spotPerKwh` price)

## What is a spike?

First of all: A spike is related to the `spotPerKwh` - spikes are per state, not per user.

For a forecast interval, it is a "spike" if `spotPerKwh` is over $3 - the first period is $15.89 (spike), the second is $1.03 (no spike)

For the current interval, it is a "spike" if the weighted average of the all the 5 minute actual and 5 minute forecasts is greater than $3 AND there has been at least one 5-minute interval over $3 (If the last predicated isn't met, the `spikeStatus` is 'predicted').

We hide all the 5-minute intervals from you, but you can infer roughly what is going on by the `spikeStatus` field.

The range... isn't the same in both cases? 

Note: $15 (ex GST) is the maximum price allowed by the NEM, so the max price described in both instances is basically as high as they go - this is pretty common.

**Reply by oldwindmill** [reply]
well, 8.1 or 8.4 as the lower limit to a top of 1600 is close... :) 
especially for peak, its below network cost. fingers croseed!

**Reply by madpilot** [reply]
General consensus is we won't spike this afternoon, but prices will probably settle high (in the $2-$3 range) for a couple of intervals.
